src/fromager/settings.py

```diff
@@ -82,6 +82,15 @@ def resolver_include_sdists(
             resolve_dist.get("include_sdists"), default
         )
 
+    def build_dir(self, pkg: str, sdist_root_dir: pathlib.Path) -> pathlib.Path:
+        p = self.get_package_settings(pkg)
+        if p.get("build_dir"):
+            # this ensures that the build_dir path from settings is relative to sdist_root_dir
+            input_build_dir = pathlib.Path(p.get("build_dir"))
+            relative_build_dir = input_build_dir.relative_to(input_build_dir.anchor)
```

Comment:
I think we're going to need some unit tests to verify the cases for this. I was able to get it to throw an exception:

```
>>> pathlib.Path('/var/tmp').relative_to(pathlib.Path('bah/humbug').anchor)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.9/pathlib.py", line 939, in relative_to
    raise ValueError("{!r} is not in the subpath of {!r}"
ValueError: '/var/tmp' is not in the subpath of '' OR one path is relative and the other is absolute.
```

Reply:
oh both the paths should be the same this example. I will still add some unit tests